sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1530,6 +1689,19 @@ class Analyzer(
   object GlobalAggregates extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
       case Project(projectList, child) if containsAggregates(projectList) =>
+
+        def traversePlanAndCheck(plan: LogicalPlan, inRecursiveTable: Boolean = false): Unit =
+          plan match {
+            case RecursiveReference(name, _) if !inRecursiveTable =>
+              throw new AnalysisException(s"Recursive reference ${name} can't be used in an " +
+                s"aggregate")
+            case RecursiveTable(_, _, recursiveTerm, _) =>
+              traversePlanAndCheck(recursiveTerm, true)
```

Comment:
my understanding is that in this case we do nothing, so can we avoid traversing the plan?

Reply:
thanks, fixed